### PR TITLE
[feature] support for new CIDv1

### LINKED ISSUE #8104: bzz://theswarm.eth doesn't work
# Problem
bzz://theswarm.eth doesn't resolve

# Expected behavior
It resolves, probably using Swarm gateway

OR link to issue on "We don't yet have Swarm support, we'd love to, here's the <GHI> you can follow along"

## Notes
Customer feedback https://www.reddit.com/r/ethereum/comments/bjhptq/status_release_0120_introducing_discover_v1_and/

cc @mandrigin @flexsurfer @yenda @annadanchenko fyi

### PR DESCRIPTION
**Pull request by yenda**
fixes #8104 

### Steps to test

Tested on emulator, open theswarm.eth page should open

status: ready

### CODE REVIEW

**Review by flexsurfer — APPROVED** [review]

**Review by vitvly — APPROVED** [review]

File: src/status_im/utils/contenthash.cljs [hunk 1]
```diff
@@ -23,17 +23,34 @@
...
-     :hash  (-> hex
-                (subs 10)
-                hex/decode
-                b58/encode)}))
+    (cond
+      (string/starts-with? hex "0xe40101")
+      ;; assuming the content type is 2 bytes
+      {:namespace :swarm
+       :hash (-> hex
+                 (subs 10)
```
Comment (flexsurfer) [inline src/status_im/utils/contenthash.cljs]: are you sure here is 10 ?

**Comment by flexsurfer** [comment]
if you could add a few tests would be great

**Review by flexsurfer — CHANGES_REQUESTED** [review]
swarm

File: src/status_im/utils/contenthash.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/utils/contenthash.cljs]: there is additional `01` for content type

Comment (yenda) [reply src/status_im/utils/contenthash.cljs]: yes 2 bytes for content type

Comment (flexsurfer) [reply src/status_im/utils/contenthash.cljs]: so it should be 12 `0xe40101fa01` ?

**Comment by yenda** [comment]
@flexsurfer fixed swarm and added test

**Comment by yenda** [comment]
@flexsurfer actually I decided to split on the hash bytes so that we don't depend on the content-type length

File: src/status_im/utils/contenthash.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/utils/contenthash.cljs]: no fa01 is just one kind of contenthash

**Review by flexsurfer — APPROVED** [review]

**Comment by yenda** [comment]
@flexsurfer theswarm.eth should open now

File: src/status_im/ethereum/ens.cljs [hunk 2]
```diff
@@ -105,8 +105,12 @@
     :method "contenthash(bytes32)"
     :params [(namehash ens-name)]
     :on-success
-    (fn [hash]
-      (cb hash))}))
+    (fn [raw-hash]
+      ;; NOTE: it would be better if our abi-spec/decode was able to do that
+      (let [hash (subs raw-hash 130)
```
Comment (flexsurfer) [inline src/status_im/ethereum/ens.cljs:110]: does it work the same for ipfs and swarm?

Comment (yenda) [reply src/status_im/ethereum/ens.cljs:110]: yes it's not decoding the contenthash itself at this point, just extracting it from the bytes soup returned by the contract

**Review by flexsurfer — APPROVED** [review]

**Pull request merged**